### PR TITLE
FAT-1454: Implement e-2-e automation of test case C2268: Add notice p…

### PR DESCRIPTION
**Pull request by pavelspichonak**
## Purpose
Implement e-2-e automation of test case C2268: Add notice policy to circulation rules

## Refs
https://issues.folio.org/browse/FAT-1454
https://foliotest.testrail.io/index.php?/cases/view/2268

### CODE REVIEW

**Review by Stanislau-Snapok — APPROVED** [review]

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import {
+  CodeMirror,
+  CodeMirrorHint,
+  Button,
+} from '../../../../../interactors';
+import InteractorsTools from '../../../../support/utils/interactorsTools';
+
+describe('ui-circulation-settings: Edit circulation rules', () => {
+  const calloutMessages = {
+    CIRCULATION_RULES_UPDATE_SUCCESS: 'Rules were successfully updated.',
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:16]: move fragment's specific into appropriate fragment

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  };
+  const newNoticePolicyData = {
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:18]: move such test data into related fragments

File: cypress/support/api/circulation.js [hunk 3]
```diff
@@ -20,3 +20,25 @@ Cypress.Commands.add('createItemCheckinApi', (data) => {
...
+Cypress.Commands.add('editCirculationRulesApi', (data) => {
+  cy
+    .okapiRequest({
+      method: 'PUT',
+      path: 'circulation/rules',
+      body: data,
+    });
+});
+
+Cypress.Commands.add('getCirculationRulesApi', () => {
```
Comment (dmitriyfolio) [inline cypress/support/api/circulation.js:33]: move api methods to related fragments

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    cy.deleteLoanPolicyApi(newLoanPolicyData.id);
+    cy.deleteRequestPolicyApi(newRequestPolicyData.id);
+    cy.deleteLostItemFeePolicyApi(newLostItemFeePolicyData.id);
+    cy.deleteOverdueFinePolicyApi(newOverdueFinePolicyData.id);
+    cy.deleteUser(newUserId);
+  });
+
+  it('C2268: Add notice policy to circulation rules', { tags: [TestType.smoke] }, () => {
+    cy.do([
+      CodeMirror().clear(),
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:117]: redesign to wrappers related with folio fragments. All interactions with web elements should be wrapped and encapsulated into fragments

Comment (pavelspichonak) [reply cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js]: done

File: cypress/support/api/circulation.js (hunk 3 above)
Comment (pavelspichonak) [reply cypress/support/api/circulation.js:165]: Do you mean in some fragment we need to add some method which uses `getCirculationRulesApi` command? Could you please explain why we need it?

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js (hunk 2 above)
Comment (pavelspichonak) [reply cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:10]: Do you mean just move it to fragment files and import these data from them?

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js (hunk 1 above)
Comment (pavelspichonak) [reply cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js]: done

File: cypress/support/api/circulation.js (hunk 3 above)
Comment (dmitriyfolio) [reply cypress/support/api/circulation.js:165]: let's imagine there is a fragment with list and button to add some object into the table. It logically to have gui and api methods related with adding of such objects into the system. In smoke tests we interact with folio fragments. And if we need to do something with folio object, we should start to work with method of related fragment. I hope this comment will be useful.

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js (hunk 2 above)
Comment (dmitriyfolio) [reply cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:10]: yes

File: cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+      CirculationRules.getCirculationRulesApi().then((circulationRules) => {
+        originalCirculationRules = circulationRules;
+      });
+      MaterialTypes.createMaterialTypeApi();
+      NoticePolicy.createNoticePolicyApi();
+      LoanPolicy.createLoanPolicyApi();
+      RequestPolicy.createRequestPolicyApi();
+      LostItemFeePolicy.createLostItemFeePolicyApi();
+      OverdueFinePolicy.createOverdueFinePolicyApi();
```
Comment (dmitriyfolio) [inline cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js:48]: fix naming. For example - OverdueFinePolicy.createOverdueFinePolicyApi(); -> OverdueFinePolicy.createApi();

**Review by dmitriyfolio — APPROVED** [review]

File: cypress/support/fragments/inventory/materialType/materialTypesSettings.js [hunk 6]
```diff
@@ -1,6 +1,15 @@
+import uuid from 'uuid';
+
 import { Button, Pane, TextField, HTML, including, MultiColumnListRow } from '../../../../../interactors';
 import ModalDeleteMaterialType from './modalDeleteMaterialType';
 import InteractorsTools from '../../../utils/interactorsTools';
+import { getTestEntityValue } from '../../../../support/utils/stringTools';
```
[the remark this answers is not in the document]
Comment (pavelspichonak) [reply cypress/support/fragments/inventory/materialType/materialTypesSettings.js]: Done

File: cypress/support/api/auth.js [hunk 7]
```diff
@@ -23,3 +23,7 @@ Cypress.Commands.add('setUserPassword', (userCredentials) => {
       body: userCredentials,
     });
 });
+
+Cypress.Commands.add('getAdminToken', () => {
```
[the remark this answers is not in the document]
Comment (pavelspichonak) [reply cypress/support/api/auth.js:27]: Done

**Pull request merged**